code/modules/client/preferences.dm

```diff
@@ -182,14 +182,23 @@ var/const/MAX_SAVE_SLOTS = 8
 /datum/preferences/New(client/C)
 	client=C
 	if(istype(C))
-		if(!IsGuestKey(C.key))
-			var/load_pref = load_preferences_sqlite(C.ckey)
-			if(load_pref)
-				if(load_save_sqlite(C.ckey, C, default_slot))
-					return
-		randomize_appearance_for()
-		real_name = random_name(gender)
-		save_character_sqlite(src, C.ckey, default_slot)
+		if(!speciesinit)
+			spawn(10)
```

Comment:
Either way this should be a while loop and not a recursive function, there's absolutely no reason for it to be a recursive function when you can just delay it with a while loop